new-lamassu-admin/src/pages/Dashboard/SystemPerformance/Graphs/RefScatterplot.js

```diff
@@ -1,197 +1,363 @@
+    },
+    [NOW, dataPoints, timeFrame]
+  )
+
+  const x = d3
+    .scaleUtc()
+    .domain(periodDomains[timeFrame])
+    .range([GRAPH_MARGIN.left, GRAPH_WIDTH - GRAPH_MARGIN.right])
+
+  const y = d3
+    .scaleLinear()
+    .domain([
+      0,
+      (d3.max(data, d => new BigNumber(d.fiat).toNumber()) ?? 1000) * 1.05
+    ])
+    .nice()
+    .range([GRAPH_HEIGHT - GRAPH_MARGIN.bottom, GRAPH_MARGIN.top])
+
+  const buildBackground = useCallback(
+    g => {
+      g.append('rect')
+        .attr('x', 0)
+        .attr('y', GRAPH_MARGIN.top)
+        .attr('width', GRAPH_WIDTH)
+        .attr('height', GRAPH_HEIGHT - GRAPH_MARGIN.top - GRAPH_MARGIN.bottom)
+        .attr('fill', backgroundColor)
+    },
+    [GRAPH_MARGIN]
+  )
+
+  const buildXAxis = useCallback(
+    g =>
+      g
+        .attr(
+          'transform',
+          `translate(0, ${GRAPH_HEIGHT - GRAPH_MARGIN.bottom})`
+        )
+        .call(
+          d3
+            .axisBottom(x)
+            .ticks(dataPoints[timeFrame].tick)
+            .tickFormat(d => {
+              return d3.timeFormat(dataPoints[timeFrame].labelFormat)(
+                d.getTime() + d.getTimezoneOffset() * MINUTE
+              )
+            })
+        )
+        .call(g => g.select('.domain').remove()),
+    [GRAPH_MARGIN, dataPoints, timeFrame, x]
+  )
+
+  const buildYAxis = useCallback(
+    g =>
+      g
+        .attr('transform', `translate(${GRAPH_MARGIN.left}, 0)`)
+        .call(d3.axisLeft(y).ticks(5))
+        .call(g => g.select('.domain').remove())
+        .selectAll('text')
+        .attr('dy', '-0.25rem'),
+    [GRAPH_MARGIN, y]
+  )
+
+  const buildGrid = useCallback(
+    g => {
+      g.attr('stroke', subheaderDarkColor)
+        .attr('fill', subheaderDarkColor)
+        // Vertical lines
+        .call(g =>
+          g
+            .append('g')
+            .selectAll('line')
+            .data(buildTicks(x.domain()))
+            .join('line')
+            .attr('x1', d => 0.5 + x(d))
+            .attr('x2', d => 0.5 + x(d))
+            .attr('y1', GRAPH_MARGIN.top)
+            .attr('y2', GRAPH_HEIGHT - GRAPH_MARGIN.bottom)
+            .attr('stroke-width', 1)
+        )
+        // Horizontal lines
+        .call(g =>
+          g
+            .append('g')
+            .selectAll('line')
+            .data(
+              d3
+                .axisLeft(y)
+                .scale()
+                .ticks(5)
+            )
+            .join('line')
+            .attr('y1', d => 0.5 + y(d))
+            .attr('y2', d => 0.5 + y(d))
+            .attr('x1', GRAPH_MARGIN.left)
+            .attr('x2', GRAPH_WIDTH - GRAPH_MARGIN.right)
+        )
+        // Thick vertical lines
+        .call(g =>
+          g
+            .append('g')
+            .selectAll('line')
+            .data(
+              buildTicks(x.domain()).filter(x => {
+                if (timeFrame === 'day') return x.getUTCHours() === 0
+                return x.getUTCDate() === 1
```

Comment:
We have this same code on three places in this file. Maybe we should isolate it in separate function?

Also, it could be written as 

```
x => timeFrame === 'day' ? x.getUTCHours() === 0 : x.getUTCDate() === 1
```